cylc/flow/parsec/fileparse.py

```diff
@@ -253,22 +262,27 @@ def process_plugins(fpath, opts):
             or ``empy``.
 
     args:
-        fpath: Directory where the plugin will look for a config.
+        fpath: Path to a config file. Plugin will look at the parent
+            directory of this file.
         opts: Command line options to be passed to the plugin.
 
     Returns: Dictionary in the form:
         extra_vars = {
             'env': {},
             'template_variables': {},
-            'templating_detected': None
+            TEMPLATING_DETECTED: None
         }
     """
+    fpath = Path(fpath)
+
     # Set out blank dictionary for return:
-    extra_vars = {
-        'env': {},
-        'template_variables': {},
-        'templating_detected': None
-    }
+    extra_vars = deepcopy(EXTRA_VARS_TEMPLATE)
+
+    # Don't run this on global configs:
+    if fpath.name == 'global.cylc':
+        return extra_vars
+
+    fpath = fpath.parent
```

Comment:
It would be a little clearer to avoid redefining this variable by adding the `.parent` bit to the one usage in the code:

```diff
            plugin_result = entry_point.load()(
-                srcdir=fpath, opts=opts
+                srcdir=fpath.parent, opts=opts
            )
```